src/visualization/Turi Create Visualization/src/js/lib/elements/Plot/Plot.js

```diff
@@ -0,0 +1,166 @@
+import React, { Component } from 'react';
+import vegaEmbed from 'vega-embed';
+import {vega as vegaTooltip} from'vega-tooltip';
+import './Plot.css';
+
+var vega = require('vega');
+
+window.vega_type = vega;
+
+class TcPlot extends Component {
+
+    componentDidMount(){
+        this.addSpec(this.props.vega_spec);
+    }
+
+    checkLoadedFlag(data, $this) {
+        if($this.vegaLoading) {
+            window.setTimeout($this.checkLoadedFlag, 100, data, $this);
+        } else {
+            if($this.vegaLoading){
+                debugger;
+            }
+
+            if (!data || !data.values) {
+                debugger;
+            }
+
+            var newData = data.values.slice();
+
+            var changeSet = vega.changeset();
+
+            if(data.progress != null){
+                var progress = parseFloat(data.progress);
+                if(parseInt(progress*100, 10) >= 100){
+                    document.getElementById('progress_bar').style.width = parseInt(progress*100)+"%";
```

Comment:
This line seems repeated in the if/else cases. You can probably move it up above the `if` so it isn't repeated.